### PR TITLE
New logic for rover home dir and creation of default files

### PR DESCRIPTION
**Pull request by benc-uk**
- Upon creation of the rover home dir `$HOME/.rover/` (checked each time rover starts), default files in pkg/rover/home will be copied into that new directory 
- Contents of pkg/rover/home is embedded into rover binary
- Custom actions are now loaded from a yaml file `actions.yaml` from the rover home dir
- Custom actions can request `setupEnv: true` in their definition which will call `landingzone.SetupEnvironment()` 
- `landingzone.SetupEnvironment()` is a new function which carries out all the account checking, identity checks, fully populates the Options struct, and sets all the environmental vars. It has been split off from `prepareTerraformCAF` which was getting too large

### CODE REVIEW

File: pkg/rover/home/tflint.hcl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+config {
```
Comment (sebastus) [inline pkg/rover/home/tflint.hcl:1]: gitignore this file?

File: pkg/testing/cli.go [hunk 2]
```diff
@@ -70,6 +70,7 @@ type SPNDetails struct {
 	ServicePrincipalNames []string
 }
 
+// Config is exported for some reason?
```
Comment (sebastus) [inline pkg/testing/cli.go:73]: yes, because of tests / asserts.

**Review by sebastus — CHANGES_REQUESTED** [review]
couple of small things.

File: pkg/rover/home/tflint.hcl (hunk 1 above)
Comment (benc-uk) [reply pkg/rover/home/tflint.hcl:1]: why? it's part of our opinionated configuration for all the tools, it's what going to be copied into users' .rover directory

**Review by sebastus — APPROVED** [review]
lgtm

**Pull request merged**